wasmbinding/message_plugin.go

```diff
@@ -139,11 +146,91 @@ func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddre
 		if contractMsg.ResubmitFailure != nil {
 			return m.resubmitFailure(ctx, contractAddr, contractMsg.ResubmitFailure)
 		}
+		if contractMsg.Dex != nil {
+			data, err := m.dispatchDexMsg(ctx, contractAddr, *(contractMsg.Dex))
+			return nil, data, err
+		}
 	}
 
 	return m.Wrapped.DispatchMsg(ctx, contractAddr, contractIBCPortID, msg)
 }
 
+// TODO: add handler name as arg for logging purpose
+// fmt.Sprintf("%T: failed to execute", handler) -> fmt.Sprintf("%Ts: failed to execute", handlerName)
+func handleDexMsg[T sdk.Msg, R any](ctx sdk.Context, msg T, handler func(ctx context.Context, msg T) (R, error)) ([][]byte, error) {
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, errors.Wrapf(err, "failed to validate %T", msg)
+	}
+
+	resp, err := handler(ctx, msg)
+	if err != nil {
+		ctx.Logger().Debug(fmt.Sprintf("%T: failed to execute", msg),
+			"from_address", msg.GetSigners()[0].String(),
+			"msg", msg,
+			"error", err,
+		)
+		return nil, errors.Wrapf(err, "failed to execute %T", msg)
+	}
+
+	data, err := json.Marshal(resp)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("json.Marshal: failed to marshal %T response to JSON", resp),
+			"from_address", msg.GetSigners()[0].String(),
+			"msg", resp,
+			"error", err,
+		)
+		return nil, errors.Wrap(err, "marshal json failed")
+	}
+
+	ctx.Logger().Debug(fmt.Sprintf("%T execution completed", handler),
+		"from_address", msg.GetSigners()[0].String(),
+		"msg", msg,
+	)
+	return [][]byte{data}, nil
+}
+
+func (m *CustomMessenger) dispatchDexMsg(ctx sdk.Context, contractAddr sdk.AccAddress, dex bindings.Dex) ([][]byte, error) {
+	if dex.Deposit != nil {
```

Comment:
would be nice and clean to use a switch statement here for msg cases just as you did for queries